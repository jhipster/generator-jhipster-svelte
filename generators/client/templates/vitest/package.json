{
	"scripts": {
		"test": "vitest run --coverage",
 		"test:ui": "vitest --ui --coverage",
 		"test:watch": "vitest"
	},
	"devDependencies": {
		"@vitest/ui": "0.34.6",
 		"@vitest/coverage-v8": "1.0.0-beta.0",
		"jsdom": "22.1.0",
		"vitest" : "0.34.6",
		"vitest-sonar-reporter": "0.5.0"
	}
}

