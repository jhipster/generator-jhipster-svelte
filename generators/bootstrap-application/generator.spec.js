import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'bootstrap-application';
const BLUEPRINT_NAMESPACE = `jhipster:${SUB_GENERATOR}`;

describe('SubGenerator bootstrap-application of svelte JHipster blueprint', () => {
	describe('run', () => {
		beforeAll(async function () {
			await helpers
				.run(BLUEPRINT_NAMESPACE)
				.withJHipsterConfig()
				.withOptions({
					ignoreNeedlesError: true,
					blueprint: 'svelte',
				})
				.withJHipsterLookup()
				.withParentBlueprintLookup();
		});

		it('should succeed', () => {
			expect(result.getStateSnapshot()).toMatchSnapshot();
		});
	});
});
