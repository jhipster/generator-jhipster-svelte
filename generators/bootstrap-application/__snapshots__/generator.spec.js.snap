// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator bootstrap-application of svelte JHipster blueprint > run > should succeed 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
}
`;
